### PR TITLE
CRVS 2028 - Updating U5 Child Form

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
There's need to add parent details to children under 5 to facilitate their birth registration process. The details are similar to those captured under child registration for out of area.
The U5 registration form has been updated to reflect the additional parent details as indicated in the data dictionary under add child under N
They include;

Type of birth place
Name of birth place
Number of children born
Type of attendance at delivery
Mode of delivery
Type of pregnancy
Baby weight
Mother's name
Mother's national ID
Mother's DOB
Mother's Marital Status
Mother's Education Level
Mother's Residence
Mother's Birth Place
Father's name
Father's national ID
Father's DOB
Father's Marital Status
Father's Education Level
Father's Residence
Father's Birth Place

### CODE REVIEW

**Review by allan-on — CHANGES_REQUESTED** [review]
We should have the English texts in the properties file and also add the French translations to the French properties file to support translations for the fields added to the form.

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 1]
```diff
@@ -328,6 +357,397 @@
...
+        "type": "spinner",
+        "hint": "{{child_enrollment.step1.type_of_pregnancy.hint}}",
+        "values": [
+          "{{child_enrollment.step1.type_of_pregnancy.value[0]}}",
+          "{{child_enrollment.step1.type_of_pregnancy.value[1]}}",
+          "{{child_enrollment.step1.type_of_pregnancy.value[2]}}",
+          "{{child_enrollment.step1.type_of_pregnancy.value[3]}}",
+          "{{child_enrollment.step1.type_of_pregnancy.value[4]}}"
+        ],
+        "openmrs_choice_ids": {
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:447]: There are no keys defined for this spinner

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 2]
```diff
@@ -328,6 +357,397 @@
...
+          "Singleton": "singleton",
+          "Twin": "twin",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:449]: There are no keys defined for this spinner

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 3]
```diff
@@ -328,6 +357,397 @@
...
+        "openmrs_choice_ids": {
+          "C-Section": "c_section",
+          "Forceps": "forceps",
+          "Normal delivery": "normal_delivery",
+          "Vacuum": "vacuum",
+          "Unknown": "unknown"
+        }
+      },
+      {
+        "key": "type_of_pregnancy",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:433]: There are no keys defined for this Spinner

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 4]
```diff
@@ -328,6 +357,397 @@
...
+        "key": "mother_id",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "concept",
+        "openmrs_entity_id": "motherId",
+        "type": "edit_text",
+        "hint": "{{child_enrollment.step1.mother_id.hint}}",
+        "edit_type": "name"
+      },
+      {
+        "key": "mother_dob_entered",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:505]: There's no relevance attached to either show the date picker or  dob unknown checkbox

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 5]
```diff
@@ -328,6 +357,397 @@
...
+        },
+        "min_date": "today-120y",
+        "max_date": "today-10y",
+        "v_required": {
+          "value": "true",
+          "err": "{{child_enrollment.step1.mother_dob_entered.error}}"
+        }
+      },
+      {
+        "key": "mother_dob_unknown",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:523]: There's no relevance attached to either show the date picker or  dob unknown checkbox

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 6]
```diff
@@ -328,6 +357,397 @@
...
+        "keys": [
+          "None",
+          "Literacy",
+          "Primary",
+          "Secondary",
+          "University"
+        ],
+        "openmrs_choice_ids": {
+          "None": "none",
+          "Literacy": "",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:592]: This should have a default value

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 7]
```diff
@@ -328,6 +357,397 @@
...
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:722]: This should have a default value

File: opensrp-chw/src/crvs/assets/json.form/child_enrollment.json [hunk 8]
```diff
@@ -328,6 +357,397 @@
...
+        "hint": "{{child_enrollment.step1.father_dob_entered.hint}}",
+        "expanded": false,
+        "duration": {
+          "label": "Age"
+        },
+        "min_date": "today-120y",
+        "max_date": "today-10y"
+      },
+      {
+        "key": "father_dob_unknown",
```
Comment (allan-on) [inline opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:657]: There's no relevance attached to either show the date picker or dob unknown checkbox There's also no father_age_entered field that's supposed to show when the checkbox is selected

Comment (qaziabubakar-vd) [reply opensrp-chw/src/crvs/assets/json.form/child_enrollment.json:712]: Relevance is added

**Pull request merged**